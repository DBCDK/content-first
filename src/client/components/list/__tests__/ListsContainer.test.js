import React from 'react';
import {Lists} from '../Lists.container';
import renderer from 'react-test-renderer';

const createTestElement = id => {
  return {
    book: {
      pid: 'pid' + id,
      title: 'some title ' + id,
      creator: 'some creator' + id
    },
    links: {
      cover: '/cover' + id
    },
    description: 'some description' + id
  };
};

const createList = id => {
  return {
    id: id,
    title: 'some list',
    description: 'some description',
    list: [createTestElement(1), createTestElement(2), createTestElement(3)]
  };
};

describe('Lists Container', () => {
  test('List is rendered', () => {
    const tree = renderer.create(<Lists lists={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.lists = [createList(1), createList(2)];
    expect(tree).toMatchSnapshot();
  });
});
