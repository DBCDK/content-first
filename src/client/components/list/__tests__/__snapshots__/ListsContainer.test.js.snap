// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists Container List is rendered 1`] = `
<div
  className="lists-page"
>
  <div
    className="mb3"
  />
  <div
    className="mb3"
  />
  <div>
    <a
      className="btn btn-primary d-inline-flex align-items-center"
      href="/lister/opret"
      onClick={[Function]}
    >
      Ny liste
      <span
        className="material-icons ml4"
      >
        add
      </span>
    </a>
  </div>
</div>
`;

exports[`Lists Container List is rendered 2`] = `
<div
  className="lists-page"
  lists={
    Array [
      Object {
        "description": "some description",
        "id": 1,
        "list": Array [
          Object {
            "book": Object {
              "creator": "some creator1",
              "pid": "pid1",
              "title": "some title 1",
            },
            "description": "some description1",
            "links": Object {
              "cover": "/cover1",
            },
          },
          Object {
            "book": Object {
              "creator": "some creator2",
              "pid": "pid2",
              "title": "some title 2",
            },
            "description": "some description2",
            "links": Object {
              "cover": "/cover2",
            },
          },
          Object {
            "book": Object {
              "creator": "some creator3",
              "pid": "pid3",
              "title": "some title 3",
            },
            "description": "some description3",
            "links": Object {
              "cover": "/cover3",
            },
          },
        ],
        "title": "some list",
      },
      Object {
        "description": "some description",
        "id": 2,
        "list": Array [
          Object {
            "book": Object {
              "creator": "some creator1",
              "pid": "pid1",
              "title": "some title 1",
            },
            "description": "some description1",
            "links": Object {
              "cover": "/cover1",
            },
          },
          Object {
            "book": Object {
              "creator": "some creator2",
              "pid": "pid2",
              "title": "some title 2",
            },
            "description": "some description2",
            "links": Object {
              "cover": "/cover2",
            },
          },
          Object {
            "book": Object {
              "creator": "some creator3",
              "pid": "pid3",
              "title": "some title 3",
            },
            "description": "some description3",
            "links": Object {
              "cover": "/cover3",
            },
          },
        ],
        "title": "some list",
      },
    ]
  }
>
  <div
    className="mb3"
  />
  <div
    className="mb3"
  />
  <div>
    <a
      className="btn btn-primary d-inline-flex align-items-center"
      href="/lister/opret"
      onClick={[Function]}
    >
      Ny liste
      <span
        className="material-icons ml4"
      >
        add
      </span>
    </a>
  </div>
</div>
`;
